tests/e2e/vmgroupbasic_test.go

```diff
@@ -0,0 +1,254 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This test suite contains tests to verify behaviors of non-default vmgroup
+
+// +build runonce
+
+package e2e
+
+import (
+	"log"
+
+	admincliconst "github.com/vmware/docker-volume-vsphere/tests/constants/admincli"
+	"github.com/vmware/docker-volume-vsphere/tests/constants/properties"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/admincli"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/dockercli"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/esx"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/inputparams"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/misc"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/verification"
+	. "gopkg.in/check.v1"
+)
+
+const (
+	vmGroupName = "vg_basictest"
+)
+
+type vgBasicSuite struct {
+	config      *inputparams.TestConfig
+	volumeNames []string
+}
+
+func (s *vgBasicSuite) SetUpSuite(c *C) {
+	basicVG := "basicVGTest"
+	s.config = inputparams.GetTestConfig()
+	if s.config == nil {
+		c.Skip("Unable to retrieve test config, skipping vmgroupbasic tests.")
+	}
+	admincli.ConfigInit(s.config.EsxHost)
+	s.volumeNames = []string{inputparams.GetUniqueVolumeName(basicVG), inputparams.GetUniqueVolumeName(basicVG)}
+}
+
+func (s *vgBasicSuite) SetUpTest(c *C) {
+	// Creating non-default vmgroup only if it does not exists
+	if !admincli.IsVmgroupPresent(s.config.EsxHost, vmGroupName) {
```

Comment:
too nested. just return if Vmgroup is present.
If not then execute steps to create it. No need for nesting.